src/solver/expressions/include/antares/solver/expressions/visitors/AstGraphVisitor.h

```diff
@@ -0,0 +1,85 @@
+/*
+** Copyright 2007-2024, RTE (https://www.rte-france.com)
+** See AUTHORS.txt
+** SPDX-License-Identifier: MPL-2.0
+** This file is part of Antares-Simulator,
+** Adequacy and Performance assessment for interconnected energy networks.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the Mozilla Public Licence 2.0 as published by
+** the Mozilla Foundation, either version 2 of the License, or
+** (at your option) any later version.
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** Mozilla Public Licence 2.0 for more details.
+**
+** You should have received a copy of the Mozilla Public Licence 2.0
+** along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+*/
+#pragma once
+
+#include <ostream>
+
+#include "antares/solver/expressions/visitors/NodeVisitor.h"
+
+namespace Antares::Solver::Visitors
+{
+
+class AstGraphVisitorNotImplemented: public std::invalid_argument
+{
+public:
+    AstGraphVisitorNotImplemented(const std::string& visitor, const std::string& node);
+};
+
+class AstGraphVisitor: public NodeVisitor<void>
```

Comment:
1st one is the return type, 2nd one is the argument type
```suggestion
class AstGraphVisitor: public NodeVisitor<void, std::ostream&>
```
So `AstGraphVisitor` can go stateless.

Then
```cpp
std::ostream& operator<<(std::ostream& os, std::pair<AstGraphVisitor&, Node*> visitorExpr)
{
    beginTree(os);
    auto& [visitor, root] = visitorExpr;
    visitor.dispatch(root, os);
    endTree(os);
    return os;
}

// Caller
std::cout << std::pair(visitor, root) << std::endl;
```